app/controllers/user_controller.php

```diff
@@ -0,0 +1,77 @@
+<?php
+class UserController extends AppController
+{
+    public function registration()
+    {
+        if(isset($_SESSION['username'])) {
+            redirect('thread/index');
+        }
+
+        $page = Param::get('page_next','registration');
+        $user = new User();
+
+        switch($page) {
+            case "registration":
+                break;
+            case "registration_end":
+                $user->fname = Param::get('fname');
+                $user->lname = Param::get('lname');
+                $user->username = Param::get('username');
+                $user->email = Param::get('email');
+                $user->password = Param::get('password');
+                try {
+                    $user->register($user);
```

Comment:
you are passing an instance of itself as parameter. You can remove this and modify your method in model using $this instead since you already declared the properties above. 
```php
public function register()
 {
    if(!$user->validate()) {
        throw new ValidationException('Invalid data.');
     }

    $params = array(
        'fname'    => $this->fname,
        'lname'    => $this->lname,
        'username' => $this->username,
        'email'    => $this->email,
         'password' => hash('sha1', $this->password)
    );
```